transmit-lib/src/cmr/transmit/metadata_db.clj

```diff
@@ -162,17 +162,25 @@
 
 (defn find-latest-concept
   "Searches metadata db for the latest concept matching the given parameters. Do not throw serivce
-  excpetion, returns the status and error message in a map in case of error."
+  exception, returns the status and error message in a map in case of error."
+  ([context params concept-type]
+   (find-latest-concept context params concept-type first))
+  ;"Allow for custom filtration of the results, instead of the default (first)"
+  ([context params concept-type filter-fn]
+   (let [{:keys [status body]} (find-concepts-raw context (assoc params :latest true) concept-type)
+         status (int status)]
+     (case status
+       200 (filter-fn (map mdb2/finish-parse-concept (json/decode body true)))
+       404 (errors/throw-service-error :not-found body)
+       ;; default
+       (errors/internal-error!
+         (format "%s search failed. status: %s body: %s"
+                 (str/capitalize (name concept-type)) status body))))))
+
+(defn find-latest-concepts
```

Comment:
Will this ever return more than one since it doesn't allow passing in a filter to override the default `first` filter?

Reply:
Yes - since we we pass in the function`(fn [item] item)` as our `filter-fn`, find-latest-concept will return the latest revision of every matching concept, instead of just whichever concept happens to be `(first ..)`

Reply:
Outdated - I have swapped over to `find-concepts`, so this comment no longer applies.